assets/src/stories-editor/components/custom-video-block-edit.js

```diff
@@ -0,0 +1,317 @@
+				// previous attributes should be removed because they may be temporary blob urls
+				setAttributes( { src: undefined, id: undefined } );
+				switchToEditing();
+				return;
+			}
+
+			// sets the block's attribute and updates the edit component from the
+			// selected media, then switches off the editing UI
+			setAttributes( { src: media.url, id: media.id } );
+			this.setState( { src: media.url, editing: false } );
+		};
+
+		if ( editing ) {
+			return (
+				<MediaPlaceholder
+					icon={ <BlockIcon icon={ icon } /> }
+					className={ className }
+					onSelect={ onSelectVideo }
+					onSelectURL={ this.onSelectURL }
+					accept="video/mp4"
+					allowedTypes={ ALLOWED_MEDIA_TYPES }
+					value={ this.props.attributes }
+					notices={ noticeUI }
+					onError={ this.onUploadError }
+				/>
+			);
+		}
+
+		const videoPosterDescription = `video-block__poster-image-description-${ instanceId }`;
+
+		return (
+			<>
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className="components-icon-button components-toolbar__control"
+							label={ __( 'Edit video', 'amp' ) }
+							onClick={ switchToEditing }
+							icon="edit"
+						/>
+					</Toolbar>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Video Settings', 'amp' ) }>
+						<ToggleControl
+							label={ __( 'Loop', 'amp' ) }
+							onChange={ this.toggleAttribute( 'loop' ) }
+							checked={ loop }
+						/>
+						<MediaUploadCheck>
+							<BaseControl
+								className="editor-video-poster-control"
+							>
+								<BaseControl.VisualLabel>
+									{ __( 'Poster Image', 'amp' ) }
+								</BaseControl.VisualLabel>
+								<MediaUpload
+									title={ __( 'Select Poster Image', 'amp' ) }
+									onSelect={ this.onSelectPoster }
+									allowedTypes={ VIDEO_POSTER_ALLOWED_MEDIA_TYPES }
+									render={ ( { open } ) => (
+										<Button
+											isDefault
+											onClick={ open }
+											ref={ this.posterImageButton }
+											aria-describedby={ videoPosterDescription }
+										>
+											{ ! poster ? __( 'Select Poster Image', 'amp' ) : __( 'Replace image', 'amp' ) }
+										</Button>
+									) }
+								/>
+								{ poster && (
+									<p
+										id={ videoPosterDescription }
+										hidden
+									>
+										{
+											/* translators: %s: the poster image URL. */
+											sprintf( __( 'The current poster image url is %s', 'amp' ), this.props.attributes.poster )
+										}
+									</p>
+								) }
+								{ ! poster && (
+									<Notice
+										status="error"
+										isDismissible={ false }
+									>
+										{ __( 'A poster is required for videos in stories.', 'amp' ) }
+									</Notice>
+								) }
+							</BaseControl>
+						</MediaUploadCheck>
+					</PanelBody>
+				</InspectorControls>
+				<figure className="wp-block-video">
+					<video
+						autoPlay
+						muted
+						loop={ loop }
+						controls={ ! loop }
```

Comment:
Wondering if instead of always displaying the controls when the loop is off we could instead add a `ToggleControl` for `controls` which would be visible only if `loop` is turned off? Thoughts?

Reply:
Wasn‘t the conclusion that controls are not visible on the front end? Adding a ToggleControl for an editor-only thing doesn‘t make much sense then I think.

Reply:
Yep, but if they're not visible on the front end then why not have `controls={ ! loop }` just `false` instead?

Reply:
Explained my thinking here: https://github.com/ampproject/amp-wp/pull/2799#discussion_r303392747

Reply:
> If loop is disabled, controls are still shown since one might want to re-start the video during editing.

Reply:
No, they‘re shown in the editor to quickly replay the video. That change to the edit component has no effect on the frontend.

Reply:
Right, that's the edit component 🤦‍♀🤦‍♀🤦‍♀Ignore, please.